p2panda-sync-next/src/managers/topic_sync_manager.rs

```diff
@@ -144,38 +166,86 @@ where
             })
         };
 
-        let inner = self.inner.lock().await;
-        inner.session_topic_map.sender(session_id).map(|tx| {
+        self.session_topic_map.sender(session_id).map(|tx| {
             Box::pin(tx.clone().with(map_fn))
                 as Pin<Box<dyn Sink<ToSync<Operation<E>>, Error = Self::Error>>>
         })
     }
 
-    fn subscribe(&self) -> impl Stream<Item = FromSync<Self::Event>> + Send + Unpin + 'static {
+    fn subscribe(&mut self) -> impl Stream<Item = FromSync<Self::Event>> + Send + Unpin + 'static {
```

Comment:
I can imagine the code would benefit from being easier to read and having less lines if we would skip the `Stream` implementation part and just go for channels instead.